### PR TITLE
use prettier

### PR DESCRIPTION
**Pull request by charlierudolph**

### CODE REVIEW

File: scripts/lint-formatting.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+
+diffs=$(node_modules/.bin/prettier -l "{src,example,features,scripts,test}/**/*.js")
+if [ -z "$diffs" ]; then
+  exit 0
+fi
+
+echo >&2 "Javascript files must be formatted with Prettier. Please run:"
+echo >&2 "node_modules/.bin/prettier --write "$diffs""
+exit 1
```
Comment (charlierudolph) [inline scripts/lint-formatting.sh]: Added this script to CI to validate it. Came from the prettier [docs](https://github.com/prettier/prettier#option-3-bash-script)

File: .eslintrc.yml [hunk 2]
```diff
@@ -17,23 +17,14 @@ parserOptions:
...
   no-negated-condition: error
   no-nested-ternary: error
   no-plusplus: error
   no-shadow: error
-  no-trailing-spaces: error
   no-unneeded-ternary: [error, {defaultAssignment: false}]
   no-var: error
   object-shorthand: [error, always]
-  quotes: [error, single]
-  semi: [error, never]
```
Comment (charlierudolph) [inline .eslintrc.yml]: removed some eslint rules that are purely stylistic and either clash with or are made redundant by prettier.

**Comment by aslakhellesoy** [comment]
Glad to see this.

Keep in mind that it will probably make all other open PRs unmergeable. If there were a recipe people could follow to update their PR branches with prettier'ed code that might help.

**Comment by aslakhellesoy** [comment]
But then again, there are only 6 open PRs - amazing compared to cucumber-jvm and cucumber-ruby

**Comment by jbpros** [comment]
@aslakhellesoy yes, I was thinking about that. There are not many PRs at the moment, I guess we could do it ourselves (copy the config files, run `eslint --fix` and voilà).

**Comment by charlierudolph** [comment]
Auhh thanks for the update @jbpros. I was hoping for some way to classify config and that also lets me removal. For existing PRs, they can merge `git checkout --ours .` and then prettify the source.

**Pull request merged**